### DISCUSSION TITLE
quickpred for unordered factors

### DISCUSSION THREAD

**Discussion by LukasWallrich** [amices/mice#348] [Ideas]
I am trying to specify an efficient predictorMatrix for a model that includes categorical variables, and ran into some trouble with `quickpred()`

Currently, quickpred converts factors without warning to their internal codes and then calculates correlations, which then leads to arbitrary output in the case of unordered factors. The documentation and vignettes do not specify how factors are handled, I only found the answer by looking at the code - should there be a warning? Also, does it make sense to use the second `mincor` criterion (correlation between value of predictor and missingness of target) to decide which variables to use to predict factors? If so, it would be great if that could be specified as an option to quickpred?

**Comment by stefvanbuuren** [comment]
Thanks for your suggestions. 

- Will add a pointer to `base::data.matrix()`, the function that does the actual conversion to the documentation, and a short comment that the results may be nonsensical for unordered factors. 
- A good imputation model should also include factors that related to the missingness, so that's why `quickpred()` also looks into that correlation. Of course, we can have a separate criterion for that correlation, but the task on the user becomes more demanding, so - for all practical purposes - I've choose `mincor` as the sole criterion for both correlations. After all, it's just a quick predictor matrix setup.

**Comment by LukasWallrich** [comment]
Thanks! I now wrote a version of the function that deals with unordered factors, initially just for my personal use: https://github.com/LukasWallrich/rNuggets/blob/master/R/mice_quickpred_extension.R - basically, it dummy codes those factors and tests whether any of the dummies exceeds mincor. 

If you think it would make sense to include that into mice, I'd be happy to create a PR - just let me know.
